### DISCUSSION TITLE
Adding a brief overview/Module-level comments at the top of each important .go file

### DISCUSSION THREAD

**Discussion by ArmoredArtist** [GoMudEngine/GoMud#91] [Ideas]
It would be helpful to include a brief overview or module-level comments at the top of each important .go file. This could provide context about the purpose and functionality of each file, which might make the code easier to navigate and understand for future contributors.

**Comment by ArmoredArtist** [comment]
Could even add a .txt file to each folder with a brief description of the contents and their purpose for a overall view without opening each file.

**Reply by Volte6** [reply]
Sure, go ahead and define one that you think would be helpful. Submit it as a pull request and I'll incorporate it ;)

**Reply by Volte6** [reply]
I do put a `README.md` in files occasionally... but also it's hard to keep up on documentation, especially when what it documents is changing all the time

**Reply by ArmoredArtist** [reply]
I noticed that in the user commands folder. In hindsight I suppose I should have understood that is where the movement logic was.

**Reply by Volte6** [reply]
That's a good example of something that has some outdated information in it... I refactored a lot of stuff recenrly and some of that isn't relevant anymore, though generally it's not bad.

**Comment by Volte6** [comment]
Closing this topic for now - eventually I'll have time to expand the documentation... but probably not until development slows down and new features (or major changes to existing code) slow down.

This may make a better issue/feature for the future, but I really don't even know what all I want to document before I tackle it. I'll revisit it in 6 months or so.
